lib/rmg/TileInfo.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * TileInfo.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include "TileInfo.h"
+
+TileInfo::TileInfo():nearestObjectDistance(float(INT_MAX)), terrain()
+{
+	occupied = ETileType::POSSIBLE; //all tiles are initially possible to place objects or passages
+}
+
+float TileInfo::getNearestObjectDistance() const
+{
+	return nearestObjectDistance;
+}
+
+void TileInfo::setNearestObjectDistance(float value)
+{
+	nearestObjectDistance = std::max<float>(0, value); //never negative (or unitialized)
+}
+bool TileInfo::shouldBeBlocked() const
+{
+	return occupied == ETileType::BLOCKED;
+}
+bool TileInfo::isBlocked() const
+{
+	return occupied == ETileType::BLOCKED || occupied == ETileType::USED;
+}
+bool TileInfo::isPossible() const
```

Comment:
What is the difference between free and possible?

Reply:
free - tile must be passable
possible - tile can be blocked or passable